cl/favorites/utils.py

```diff
@@ -200,30 +252,76 @@ def send_prayer_emails(instance: RECAPDocument) -> None:
 
 
 async def get_user_prayer_history(user: User) -> tuple[int, float]:
-    filtered_list = Prayer.objects.filter(user=user, status=Prayer.GRANTED)
+    filtered_list = Prayer.objects.filter(
+        user=user, status=Prayer.GRANTED
+    ).select_related("recap_document")
 
     count = await filtered_list.acount()
 
     total_cost = 0
     async for prayer in filtered_list:
-        total_cost += float(price(prayer.recap_document))
+        doc_price = price(prayer.recap_document)
+        if not doc_price:
+            continue
+        total_cost += float(doc_price)
 
     return count, total_cost
 
 
-async def get_lifetime_prayer_stats() -> tuple[int, int, float]:
+@dataclass
+class PrayerStats:
+    prayer_count: int
+    distinct_count: int
+    total_cost: str
 
-    filtered_list = Prayer.objects.filter(status=Prayer.GRANTED)
 
-    count = await filtered_list.acount()
+async def get_lifetime_prayer_stats(
+    status: int,
+) -> (
+    PrayerStats
+):  # status can be only 1 (WAITING) or 2 (GRANTED) based on the Prayer model
 
-    total_cost = 0
-    distinct_documents = set()
+    cache_key = f"prayer-stats-{status}"
 
-    async for prayer in filtered_list:
-        distinct_documents.add(prayer.recap_document)
-        total_cost += float(await price(prayer.recap_document))
+    prayer_by_status = Prayer.objects.filter(status=status)
 
-    num_distinct_purchases = len(distinct_documents)
+    prayer_count = await prayer_by_status.acount()
+
+    distinct_prayers = (
+        await prayer_by_status.values("recap_document").distinct().acount()
+    )
 
-    return count, num_distinct_purchases, total_cost
+    total_cost = await (
+        prayer_by_status.select_related("recap_document")
+        .values("recap_document")
+        .distinct()
+        .annotate(
+            price=Case(
+                When(recap_document__is_free_on_pacer=True, then=Value(0.0)),
+                When(recap_document__page_count__gt=30, then=Value(3.0)),
+                When(
+                    recap_document__page_count__gt=0,
+                    recap_document__page_count__lte=30,
+                    then=F("recap_document__page_count") * 0.10,
+                ),
+                default=Value(0.0),
+            )
```

Comment:
I think it is possible to simplify this annotation a bit by relying on `Least`:

```suggestion
            price=Case(
                When(recap_document__is_free_on_pacer=True, then=Value(0.0)),
                When(
                    recap_document__page_count__gt=0,
                    then=Least(
                        Value(3.0),
                        F("recap_document__page_count") * Value(0.10)
                    ),
                ),
                default=Value(0.0),
            )
```

I would also suggest adding a test for this method since it is a bit complex. This will help guarantee its functionality and ease future maintainability.